### DISCUSSION TITLE
Get lambda Full account concurrency and Unreserved account concurrency

### DISCUSSION THREAD

**Discussion by pendyalal** [cloud-custodian/cloud-custodian#7355] [General]
**Is your feature request related to a problem? Please describe.**
we are unable to get lambda account level concurrency limits.

**Describe the solution you'd like**
We want to get Lambda Full account and Unreserved account concurrency limits per region in an account.  If the unreserved concurrency falls below a certain threshold, then set the concurrency to a specific value for each lambda function that has more than 100 concurrent executions in 5 minutes. This is to ensure all Lambda functions get fair reserved capacity. 

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#Lambda.Client.get_account_settings

**Comment by kapilt** [comment]
if your concurrency is exhausted, using a lambda to check it and respond after the fact isn't a good approach. why not just set an upper bound per function? and tag exempt those with needs.

**Comment by pendyalal** [comment]
@kapilt ,That's true if the reserved concurrency is set on the lambda function.
There is a concurrency executions metric for each function. If the concurrency executions is more than 100 for 5 minutes and If the lambda function doesn't have a reserved concurrency set on lambda and if the unreserved account concurrency is more than 50% of full account concurrency then we want to set concurrency to 100 on the lambda.

**Comment by pendyalal** [comment]
@kapilt , Is there a way to get unreserved account concurrency and full account concurrency using custodian policy?

**Comment by kapilt** [comment]
we probably need to expose metric filter on the account to have this available, but you also want comparisons between the two metrics.. which relates to #5625
